XbSymbolDatabase.c

```diff
@@ -658,39 +956,113 @@ void XbSymbolScanOOVPA(OOVPATable *OovpaTable,
+                xref_count += internal_OOVPA_register(pContext, pLastKnownSymbol, pLibrary, pLastKnownFunc);
                 pLastKnownSymbol = NULL;
                 pLastKnownFunc = 0;
             }
         }
 
         // Skip higher build version
-        if (bStrictBuildVersionLimit && buildVersion < pLoop->Version)
+        if (pContext->strict_build_version_limit && pLibrary->build_version < pLoop->Version)
             continue;
 
         // Search for each function's location using the OOVPA
-        xbaddr pFunc = (xbaddr)(uintptr_t)XbSymbolLocateFunction(pLoop->szFuncName, pLoop->Version, pLoop->Oovpa, lower, upper, (uintptr_t)xb_start_virt_addr);
+        xbaddr pFunc = (xbaddr)(uintptr_t)internal_LocateFunction(pContext, pLoop->szFuncName, pLoop->Version, pLoop->Oovpa, pSection, xref_first_pass);
         if (pFunc == 0) {
             continue;
         }
 
         if (pFunc == pLastKnownFunc && pLastKnownSymbol == pLoop - 1) {
             //if (g_SymbolAddresses[pLastKnownSymbol->szFuncName] == 0) {
-                XbSymbolOutputMessageFormat(XB_OUTPUT_MESSAGE_WARN, "Duplicate OOVPA signature found for %s, %hu vs %hu!", pLastKnownSymbol->szFuncName, pLastKnownSymbol->Version, pLoop->Version);
+                output_message_format(&pContext->output, XB_OUTPUT_MESSAGE_WARN, "Duplicate OOVPA signature found for %s, %hu vs %hu!", pLastKnownSymbol->szFuncName, pLastKnownSymbol->Version, pLoop->Version);
             //}
         }
 
-        if (buildVersion < pLoop->Version) {
-            XbSymbolOutputMessageFormat(XB_OUTPUT_MESSAGE_WARN, "OOVPA signature is too high for [%hu] %s!", pLoop->Version, pLoop->szFuncName);
+        if (pLibrary->build_version < pLoop->Version) {
+            output_message_format(&pContext->output, XB_OUTPUT_MESSAGE_WARN, "OOVPA signature is too high for [%hu] %s!", pLoop->Version, pLoop->szFuncName);
         }
 
         pLastKnownFunc = pFunc;
         pLastKnownSymbol = pLoop;
     }
 
     if (pLastKnownSymbol != NULL) {
-        XbSymbolRegisterOOVPA(pLastKnownSymbol, LibraryName, LibraryFlag, pLastKnownFunc, register_func);
+        xref_count += internal_OOVPA_register(pContext, pLastKnownSymbol, pLibrary, pLastKnownFunc);
     }
+
+    return xref_count;
+}
+
+static eLibraryType internal_GetLibraryType(uint32_t library)
+{
+    switch (library) {
+        default:
+            return LT_UNKNOWN;
+        case XbSymbolLib_D3D8:
+        case XbSymbolLib_D3D8LTCG:
+            return LT_D3D;
+        case XbSymbolLib_DSOUND:
+        case XbSymbolLib_XACTENG:
+            return LT_AUDIO;
+        case XbSymbolLib_JVS:
+            return LT_JVS;
+        case XbSymbolLib_XAPILIB:
+            return LT_XAPI;
+        case XbSymbolLib_XGRAPHC:
+            return LT_GRAPHIC;
+        case XbSymbolLib_XONLINE:
+        case XbSymbolLib_XONLINES:
+        case XbSymbolLib_XNET:
+        case XbSymbolLib_XNETS:
+        case XbSymbolLib_XNETN:
+            return LT_NETWORK;
+    }
```

Comment:
Remove this function, or implement it as a lookup in `SymbolDBList` (instead of switch/case code).

Reply:
Since SymbolDBList could be used two or more times in a single library scan. This function, `internal_GetLibraryType` will stay to keep the scan process active once. 😉